README.md

```diff
@@ -85,12 +117,415 @@ return (
+
+const {controller} = createOnboarding({
+  /**/
+  plugins: [
+    new MultiTabSyncPlugin({enableStateSync: false}),
+    new PromoPresetsPlugin(),
+    new WizardPlugin(),
+  ],
+});
+```
+
+---
+
+You can write your own plugin
+
+```typescript jsx
+import {createOnboardin} from '@gravity-ui/onboarding';
+import {WizardPlugin} from '@gravity-ui/onboarding/dist/plugins';
+
+const myPlugin = {
+  apply: (onboarding) => {
+    /**
+     * Do something with onboarding controller
+     * For exampe subscribe on event
+     *  onboarding.events.subscribe('init', () => {});
+     */
+  },
+};
+
+const {controller} = createOnboarding({
+  /**/
+  plugins: [new WizardPlugin(), myPlugin],
+});
+```
+
+# Promo manager
+
+## How to use promo-manager
+### 1. Init promo manager and setup the progress update
+
+```typescript jsx
+// promo-manager.ts
+
+import { createPromoManager } from '@gravity-ui/onboarding/dist/promo-manager';
+import { ShowOnceForPeriod } from '@gravity-ui/onboarding/dist/promo-manager/helpers';
+
+export const { controller, usePromoManager, useActivePromo } = createPromoManager({
+    config: {
+        promoGroups: [{
+            slug: 'poll',
+            conditions: [ShowOnceForPeriod({month: 1})],
+            promos: [
+                {
+                    slug: 'issuePoll',
+                    conditions: [ShowOncePerMonths(6)],
+                    meta: {...}
+                },
+            ],
+        }],
+    },
+    progressState: () => {/* ... */},
+    getProgressState: () => {/* ... */},
+    onSave: {
+        progress: (state) => () => {/* ... */},
+    },
+});
+```
+
+### 2. Trigger promo in your component
+
+```typescript jsx
+// TriggerExample.tsx
+
+import { usePromoManager } from './promo-manager';
+
+const { status, requestStart, cancelStart } = usePromoManager('issuePoll');
+
+useMount(() => {
+    requestStart();
+});
+
+useUnmount(() => {
+    cancelStart();
+});
+
+if(status === 'active') {
+    // allowed to run. Do something
+}
+```
+
+## Condition and constraints
+You can use conditions for each promo. Or use constraints to set limitations between promos.
+```typescript jsx
+import {
+    ShowOnceForSession,
+    ShowOnceForPeriod,
+    MatchUrl,
+    LimitFrequency
+} from '@gravity-ui/onboarding/dist/promo-manager/helpers';
+
+const groupOfPolls = {
+    slug: 'groupOfPolls',
+    promos: [
+        {slug: 'somePoll', conditions: [ShowOnceForPeriod({month: 1})]},
+        {slug: 'pollForPageWithparam', conditions: [
+            MatchUrl('param=value'),
+            ShowOnceForPeriod({month: 5})
+        ]}
+    ],
+}
+
+const groupOfHints = {
+    slug: 'groupOfHints',
+    conditions: [ShowOnceForSession()],
+    promos: [
+        {slug: 'someHint'},
+        {slug: 'hintForSpecificPage', conditions: [
+            MatchUrl('/folder/\\w{5}/page$'),
+        ]},
+    ],
+}
+
+const {controller} = createPromoManager({
+    config: {
+        constraints: [
+            LimitFrequency({
```

Comment:
How conditions in `groupOfPolls`, `groupOfHints` and separate promos correlate with each other and `LimitFrequency`.

Reply:
Promo could be started only if
- All promo conditions returns true
- All promo group condition returns true
- All constraints passed
(Added to readme)